test/solidity/yul/transaction-information/TransactionInfo.js

```diff
@@ -0,0 +1,164 @@
+    )
+
+    expect(result).to.eq(expectedSignerABalance)
+  })
+
+  it('Should get self balance', async () => {
+    const expectedSelfBalance = Math.round(
+      INITIAL_BALANCE / tinybarToWeibarCoef
+    )
+    const result = await transactionInfoContract.getSelfBalance()
+
+    expect(result).to.eq(expectedSelfBalance)
+  })
+
+  it('Should get message caller', async () => {
+    const expectedMessageCaller = signers[0].address
+
+    const result = await transactionInfoContract.getMsgCaller()
+
+    expect(result).to.eq(expectedMessageCaller)
+  })
+
+  it('Should get message call value', async () => {
+    const expectedValue = 10_000_000_000
+
+    const transaction = await transactionInfoContract.getCallValue({
+      value: expectedValue,
+    })
+    const receipt = await transaction.wait()
+
+    const event = receipt.events.map((e) => e.event === 'CallValue' && e)[0]
+
+    const [messageValue] = event.args
+
+    expect(messageValue).to.eq(expectedValue / tinybarToWeibarCoef)
+  })
+
+  it('Should get message call data', async () => {
+    const index = 2
+    const functionSig = 'getCallDataLoad(uint256)'
+    const callData = transactionInfoContract.interface
+      .encodeFunctionData(functionSig, [index])
+      .replace('0x', '')
+
+    // @notice since transactionInfoContract.getCallDataLoad() returns the msg.calldata from memory offset `index`,
+    //         `bytes32CallData` also needs to dynamically truncate itself based on `index`
+    const expectedBytes32CallData =
+      `0x` + callData.slice(index * 2, 64 + index * 2)
+
+    const result = await transactionInfoContract.getCallDataLoad(index)
+
+    expect(result).to.eq(expectedBytes32CallData)
+  })
+
+  it('Should get the size of message call data', async () => {
+    const messagecallData = await transactionInfoContract.getCallDataLoad(0)
+    const callDataBytesArraay = ethers.utils.arrayify(messagecallData)
+    const significantBytesLength = callDataBytesArraay.reduce(
+      (length, byte) => {
+        if (byte !== 0) {
+          return (length += 1)
+        } else {
+          return length
+        }
+      },
+      0
+    )
+
+    const result = await transactionInfoContract.getCallDataSize()
+
+    expect(result).to.eq(significantBytesLength)
+  })
+
+  it('Should copy message call data to memory', async () => {
+    const dataPosF = 0
+    const memPosT = 0x20
+    const bytesAmountS = 4 // max amount
+    const functionSig = 'callDataCopier(uint256, uint256, uint256)'
+
+    const messageCallData = transactionInfoContract.interface
+      .encodeFunctionData(functionSig, [memPosT, dataPosF, bytesAmountS])
+      .replace('0x', '')
+
+    const bytes32MessageCallData =
+      '0x' + messageCallData.slice(dataPosF * 2, 64 + dataPosF * 2)
+
+    const result = await transactionInfoContract.callDataCopier(
+      memPosT,
+      dataPosF,
+      bytesAmountS
+    )
+
+    expect(result).to.eq(bytes32MessageCallData)
+  })
+})
```

Comment:
Nit. Extra line.

Reply:
Thanks for the comment @ebadiere but that file actually already has an extra line but GitHub doesn't show the extra line. But if there is not an extra blank line at the end, it would show this symbol like this

<img width="153" alt="image" src="https://github.com/hashgraph/hedera-smart-contracts/assets/66233296/574c4d74-2df6-48a0-be58-dbefafedf794">